#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

gen.add("min_x", double_t, 0, "minimum x range of heightmap [m]", -2.0, -40.0, 40.0)
gen.add("max_x", double_t, 0, "maximum x range of heightmap [m]", 2.0, -40.0, 40.0)
gen.add("min_y", double_t, 0, "minimum y range of heightmap [m]", -2.0, -40.0, 40.0)
gen.add("max_y", double_t, 0, "maximum y range of heightmap [m]", 2.0, -40.0, 40.0)
gen.add("min_z", double_t, 0, "minimum z range of heightmap [m]", -50.0, -50.0, 50.0)
gen.add("max_z", double_t, 0, "maximum z range of heightmap [m]", 50.0, -50.0, 50.0)
gen.add("resolution_x", int_t, 0, "resolution of heightmap in x axis", 400, 1, 4096)
gen.add("resolution_y", int_t, 0, "resolution of heightmap in y axis", 400, 1, 4096)
gen.add("initial_probability", double_t, 0, "initial probability to set", 1.0, 0.0, 1.0)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "HeightmapConverter"))
